addons/uh60_flir/functions/fnc_setup.sqf

```diff
@@ -1,52 +1,171 @@
+
+private _OpticsIn = _pilotCameraConfig >> "OpticsIn";
 private _fovClasses = "true" configClasses _OpticsIn;
-private _fovObjects = _fovClasses apply {
+vtx_uh60_flir_OpticsInfo = createHashMapFromArray (_fovClasses apply {
 	[
-		getNumber (_x >> "initFov"),
-		getArray (_x >> "visionMode"),
-		getArray (_x >> "thermalMode"),
-		getText (_x >> "opticsDisplayName")
+    getNumber (_x >> "initFov"),
+		[
+      getArray (_x >> "visionMode"),
+  		getArray (_x >> "thermalMode"),
+  		getText (_x >> "opticsDisplayName")
+    ]
 	]
+});
+vtx_uh60_flir_camFOVLevels = keys vtx_uh60_flir_OpticsInfo;
+vtx_uh60_flir_camFOVLevels sort false; // Descending
+vtx_uh60_flir_FOV = vtx_uh60_flir_camFOVLevels # 0;
+
+vtx_uh60_flir_camPos = getPilotCameraPosition _vehicle;
+private _dir = getPilotCameraDirection _vehicle;
+vtx_uh60_flir_camDirAndUp = [
+  _dir,
+  _dir vectorCrossProduct (_dir vectorCrossProduct [0, 0, -1])
+];
+
+vtx_uh60_flir_isVisibleMap = visibleMap;
+call vtx_uh60_flir_fnc_setIsPipHidden;
+
+if (vtx_uh60_flir_playerIsPilot) then {
+  if (productVersion # 2 >= 207) then {
+    // https://community.bistudio.com/wiki/Arma_3:_Event_Handlers#VisionModeChanged
+    _id = player addEventHandler ["VisionModeChanged", {
+    	params ["", "_visionMode", "_TIindex"];
+      if (cameraView == "GUNNER") then {
+        [vtx_uh60_flir_visionModesHashMap get [_visionMode, _TIindex]] call vtx_uh60_flir_fnc_setVisionMode;
+      };
+    }];
+    vtx_uh60_flir_playerEHs pushBack ["VisionModeChanged", _id];
+  } else {
+    _id = ["visionMode", {
+      params ["_unit", "_newvVsionMode", "_oldVisionMode"];
+      if (cameraView == "GUNNER") then {
+        // Sync Vision Mode
+        [vtx_uh60_flir_visionModesHashMap get [_newvVsionMode, [-1, 0] select (_newvVsionMode > 1)]] call vtx_uh60_flir_fnc_setVisionMode;
+      };
+    }] call CBA_fnc_addPlayerEventHandler;
+    vtx_uh60_flir_playerCBAEHs pushBack ["visionMode", _id];
+  };
 };
-private _mempoint = getText (_vehicleConfig >> "memoryPointDriverOptics");
-
-if (_mempoint == "slingcam") exitWith {false};
-
-_vehicle setVariable ["vtx_flir_mempoint", _mempoint];
-_vehicle setVariable ["vtx_flir_turnLimits", [
-	getNumber (_pilotCameraConfig >> "minTurn"),
-	getNumber (_pilotCameraConfig >> "maxTurn"),
-	getNumber (_pilotCameraConfig >> "minElev"),
-	getNumber (_pilotCameraConfig >> "maxElev")
-]];
-_vehicle setVariable ["vtx_flir_fovObjects", _fovObjects];
-_vehicle setVariable ["vtx_flir_initFovMode", 0];
-_vehicle setVariable ["vtx_flir_initVisionMode", 0];
-
-vtx_uh60_flir_rightMouseDown = false;
-[] spawn {
- 	waitUntil {!(isNull (findDisplay 46))};
-	vtx_uh60_flir_mouseButtonEvent = findDisplay 46 displayAddEventHandler ["MouseButtonDown", "_this call vtx_uh60_flir_fnc_handleKeyInputs;"];
-};
 
-vtx_uh60_flir_stabilizing = false;
-vtx_uh60_flir_stabilized = false;
-vtx_uh60_flir_zoomChanging = false;
-vtx_uh60_flir_visionChanging = false;
-vtx_uh60_flir_enteringOptics = false;
+// track SACLOS
```

Comment:
There should not be missile guidance code in the FLIR module.

Reply:
Do  you want this in weapons? It's kind of directly related to the scripted camera.

Reply:
Normally I'd put it in the weapons and apply the script to this specific rocket.
In this case I'd suggest completely removing it and I'll just make us an APKWS rocket instead so we have no SACLOS at all anymore

Reply:
7f917a8